tests/unit/src/util/Travis.ts

```diff
@@ -0,0 +1,264 @@
+import * as assert from 'intern/chai!assert';
+import loadModule, { cleanupModuleMocks } from '../../../_support/loadModule';
+import { stub, SinonStub } from 'sinon';
+import Travis, { Repository } from 'src/util/Travis';
+
+let module: any;
+let travis: Travis;
+let repository: Repository;
+let requestStub: SinonStub & Partial<{ get: SinonStub, post: SinonStub }>;
+
+registerSuite({
+	name: 'util/Travis',
+
+	before() {
+		requestStub = stub();
+		requestStub.post = stub();
+		requestStub.get = stub();
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		module = loadModule('src/util/Travis', {
+			'@dojo/core/request': { default: requestStub }
+		}, false);
+	},
+
+	afterEach() {
+		requestStub.reset();
+		requestStub.post.reset();
+		requestStub.get.reset();
+	},
+
+	'Travis': {
+		beforeEach() {
+			travis = new module.default();
+		},
+
+		async authenticate() {
+			const token = 'token';
+			const accessToken = 'access_token';
+			const post = requestStub.post;
+
+			post.returns(Promise.resolve({ json: () => Promise.resolve({ 'access_token': accessToken }) }));
+
+			const authenticate = await travis.authenticate(token);
+
+			assert.strictEqual(post.lastCall.args[1].body, '{"github_token":"token"}');
+			assert.strictEqual(travis.token, accessToken);
+			assert.strictEqual(authenticate, accessToken);
+		},
+
+		createAuthorization: (() => {
+			let repo: any;
+			const tokenAuth = { token: 'token' };
+
+			return {
+				beforeEach() {
+					repo = {
+						createAuthorization: stub().returns(Promise.resolve(tokenAuth)),
+						findAuthorization: stub().returns(Promise.resolve())
+					};
+				},
+
+				afterEach() {
+					repo.createAuthorization.reset();
+					repo.findAuthorization.reset();
+				},
+
+				async 'existing authorization; eventually throws'() {
+					repo.findAuthorization.returns(Promise.resolve({ id: 1 }));
+
+					try {
+						await travis.createAuthorization(repo);
+
+						assert.fail();
+					} catch (e) {
+						assert.strictEqual(e.message, 'An existing authorization exists. "#1"');
+						assert.isTrue(repo.createAuthorization.notCalled);
+						assert.isTrue(repo.findAuthorization.calledOnce);
+					}
+				},
+
+				async 'authentication succeeds'() {
+					const auth = stub(travis, 'authenticate').returns(Promise.resolve());
+					const deleteAuth = stub(travis, 'deleteAuthorization');
+
+					await travis.createAuthorization(repo);
+
+					assert.isTrue(repo.createAuthorization.calledOnce);
+					assert.isTrue(auth.calledOnce);
+					assert.isTrue(deleteAuth.notCalled);
+					assert.isTrue(travis.isAuthorized());
+
+					auth.restore();
+					deleteAuth.restore();
+				},
+
+				async 'authentication fails; eventually throws'() {
+					const auth = stub(travis, 'authenticate').returns(Promise.reject());
+					const deleteAuth = stub(travis, 'deleteAuthorization').returns(Promise.resolve());
+
+					try {
+						await travis.createAuthorization(repo);
```

Comment:
This test passes if `#createAuthorization()` doesn't throw. We should test for that case as well.